openshift-kube-apiserver/admission/storage/csiinlinevolumesecurity/admission.go

```diff
@@ -0,0 +1,293 @@
+package csiinlinevolumesecurity
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/apiserver/pkg/admission"
+	"k8s.io/apiserver/pkg/admission/initializer"
+	"k8s.io/apiserver/pkg/audit"
+	"k8s.io/apiserver/pkg/warning"
+	"k8s.io/client-go/informers"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	storagev1listers "k8s.io/client-go/listers/storage/v1"
+	"k8s.io/component-base/featuregate"
+	"k8s.io/klog/v2"
+	appsapi "k8s.io/kubernetes/pkg/apis/apps"
+	batchapi "k8s.io/kubernetes/pkg/apis/batch"
+	coreapi "k8s.io/kubernetes/pkg/apis/core"
+	"k8s.io/kubernetes/pkg/features"
+	podsecapi "k8s.io/pod-security-admission/api"
+)
+
+const (
+	// Plugin name
+	PluginName = "storage.openshift.io/CSIInlineVolumeSecurity"
+	// Label on the CSIDriver to declare the driver's effective pod security profile
+	csiInlineVolProfileLabel = "security.openshift.io/csi-ephemeral-volume-profile"
+	// Default values for the profile labels when no such label exists
+	defaultCSIInlineVolProfile  = podsecapi.LevelPrivileged
+	defaultPodSecEnforceProfile = podsecapi.LevelRestricted
+	defaultPodSecWarnProfile    = podsecapi.LevelRestricted
+	defaultPodSecAuditProfile   = podsecapi.LevelRestricted
+)
+
+var (
+	podSpecResources = map[schema.GroupResource]bool{
+		coreapi.Resource("pods"):                   true,
+		coreapi.Resource("replicationcontrollers"): true,
+		coreapi.Resource("podtemplates"):           true,
+		appsapi.Resource("replicasets"):            true,
+		appsapi.Resource("deployments"):            true,
+		appsapi.Resource("statefulsets"):           true,
+		appsapi.Resource("daemonsets"):             true,
+		batchapi.Resource("jobs"):                  true,
+		batchapi.Resource("cronjobs"):              true,
```

Comment:
I'd expect this list include DeploymentConfigs, not sure if Buillds and BuildConfigs is also a thing you consider?

Reply:
Apparently this can only be done from `openshift-apiserver` for OCP workloads (I didn't realize this). I discussed with @deads2k and dropped the commit I added for those workloads because the admission plugin was never getting those requests, and the resources in this list are consistent with the PodSecurity plugin.